This patch is from OpenSUSE .src.rpm for the following crash on startup:

ERROR:compositelookup.cpp:53:void CompositeLookup::send_net_dict_request(const string&, const string&): assertion failed: (NetDictRequests.end() == std::find(NetDictRequests.begin(), NetDictRequests.end(), request))

--- dict/src/lib/compositelookup.cpp
+++ dict/src/lib/compositelookup.cpp
@@ -50,8 +50,10 @@
 void CompositeLookup::send_net_dict_request(const std::string& dict_id, const std::string& key)
 {
 	NetDictRequest request(dict_id, key);
-	g_assert(NetDictRequests.end() == std::find(NetDictRequests.begin(), NetDictRequests.end(), request));
-	NetDictRequests.push_back(request);
+	if(NetDictRequests.end() == std::find(NetDictRequests.begin(), NetDictRequests.end(), request))
+	{
+	  NetDictRequests.push_back(request);
+	}
 }
 
 /* returns true if got expected response */
